New-About-Me-V1: (WIP) Let's make some memories.

ChristopherDAdams opened the pull request:
## About The Pull Request
Detailed Overview: (Bare With Me.)
This PR Includes a host of new features for TFN. It centralizes character information displays, standardizes display for each species as needed, handles player run groups/parties, allows for city-wide/group or personal chronicles, manages personal and group relationships, and stores all of it, as well shared or personal memories in one self-contained package. All in a central subsystem intended for persistence of key details. For now, it is built **ONLY** on a round-to-round basis, meaning character design, and role inform all of the starting conditions or round join conditions. With all that said, adding more features is being put on hold to test the current features and polish what is already made, and warm up the community what this can be used for.

This PR is being designed to have each part work independently, allowing for multiple areas to be worked on at once. I very much encourage others to participate and look at groups_canon.dm, and the organizations there-in, and I would love to honor the work many of you have put into these communities.

Dynamically generated on round start, or join, and managed by players and staff. This PR is Designed to keep in mind future persistence features, such as persisting blood hunts, rivalries, boons, blood-bonds, and much more. HOWEVER, these features are not supported yet. **Right now, Groups are the focus, and managing groups on an RP basis, in a way that facilitates control, but doesn't break immersion.** So RP should remain at a round to round basis, based on role and the currently supported game features. This is the core needed to support that kind of control and persistence, so many features may not be fully balanced, or are intended to be used only when RP demands it. 

Use features with care for their intent and RP purposes. Otherwise, features will be more limited as they appear, hold each other accountable, report any forms of abuses of this system. As well, I certainly appreciate detailed feedback with alternatives for how things can be handled.


**Feature Breakdown!**
-  **About Me TGUI Panel**: New UI panel with five tabs — *Overview, Groups, Relationships, Chronicle, Memories*.
-  **About Me Component**: Auto-attached to all human mobs, only works with a client, acts as the character's access point to the subsystem, through the TGUI interface, acts their character's "account". NPCS/Late-Join/Guests/est joins, etc, should never be save supported.
-  **About Me Record**: This is generated by a 'human' mob with a client, when it opens the about me UI, and acts as the key holder for the player's groups relationships chronicles and memories, held by the subsystem, accessed by the component. Will be used to save and load data and datum keys, and other persistent RP data.
-  **Groups System**: Clans, sects, factions, coteries, and more — all tracked with members, officers, leaders, and types. Loyalty to a group relationship, is controlled by both the members and the leadership. With some features to leave or rebel within groups dynamically.
- **Relationships**: Dynamic group/player relationships visible in the UI.
-  **Chronicle System**: Player-logged and shared events (e.g. wars, deals, betrayals) tied to groups and personal memories.
-  **Memory System**: Player driven personal "log-entries" that affect their loyalties, tags, and history. Created personally, or in response to external influences.
-  **SSrpmanagement Subsystem**:
  - Initializes canonical groups at round start.
  - Centralizes group, chronicle, relationships, and memory datum states.
  - Controls safe access to faction/player data.
-  Full modular backend (`aboutme_core.dm`, `aboutme_tgui.dm`, `memory.dm`, etc.)

##  Why It's Good For The Game
This system creates a **living memory** of your character’s journey, for now just on a round by round basis.

Players Features:
- Shape their character identity and story with more dynamic and centralized tools. Using about-me UI as an in-round access.
- Track and control their character's involvement in groups, chronicles, relationships, and memories.
- RP each of these features to achieve new stories, and have tools to reflect that in game.
- Create their own coteries, parties, or squads. Player run parties can be very dynamic, filling in for organizations that aren't yet implemented, or creating sub-groups of players that could "blur the lines" a bit for groups.

Staff Oversight Features:
- Tools to oversee the system, groups, the other datums, and to manage records.
- Interactions with the system to correct issues, update citywide group states, admin logging of harsher features, etc.
- Ability to turn off features if needed, while leaving other core functions intact.


##  Testing Photographs and Procedure

<details>
<summary>📸 Screenshots & Video</summary>

 **About Me Panel (Overview Tab)**  
<img width="405" height="728" alt="image" src="https://github.com/user-attachments/assets/f4a78c78-88d4-40f5-9ba6-53a8b949c0a9" />
<img width="321" height="333" alt="image" src="https://github.com/user-attachments/assets/bf8d02bb-d9cc-4fe4-8d05-5359440c341a" />

 **Groups Panel** 
<img width="394" height="773" alt="image" src="https://github.com/user-attachments/assets/8aa74c96-c001-4e66-b92e-c5021bfcf9df" />
<img width="365" height="192" alt="image" src="https://github.com/user-attachments/assets/22ea4f07-2a0b-4b41-978c-832e373c1f2b" />
<img width="368" height="165" alt="image" src="https://github.com/user-attachments/assets/42fe67a8-16ba-4ccf-920d-1de0c71738ec" />
<img width="372" height="181" alt="image" src="https://github.com/user-attachments/assets/a95fa816-82a2-43bf-8f50-74ad54ad81fe" />
<img width="370" height="182" alt="image" src="https://github.com/user-attachments/assets/0e6b7019-ad26-4e4b-b404-f2fa6ec92002" />
<img width="370" height="147" alt="image" src="https://github.com/user-attachments/assets/a6ed5c2c-69cf-426d-b1ef-f29db90d7b7f" />
<img width="322" height="336" alt="image" src="https://github.com/user-attachments/assets/99577985-17fe-4a3f-bb0f-dddc542f9505" />
<img width="344" height="411" alt="image" src="https://github.com/user-attachments/assets/ce17c24d-9b24-4f1c-bb15-8949debfdfee" />

 **Relationships Tab**  
<img width="398" height="604" alt="image" src="https://github.com/user-attachments/assets/8930e16c-075a-4f42-8a68-aa013807f5df" />
<img width="323" height="337" alt="image" src="https://github.com/user-attachments/assets/aaa3c582-cdfb-4a3c-949f-1e118ffcac95" />

 **Chronicles Panel**  
<img width="396" height="766" alt="image" src="https://github.com/user-attachments/assets/f6a3af85-e2d8-4c05-9718-f1c4b2763a15" />

 **Memories Tab**  
<img width="396" height="773" alt="image" src="https://github.com/user-attachments/assets/dc9952bb-6f0b-47e0-b17f-7697214a0c3c" />

All panels tested on:
- Human and Kindred Characters, (Garou, and Ghouls Pending)
- All TGUI buttons return valid actions via `ui_act`

</details>

## Backend Technical Stuff

<details>
<summary> Dev/Contributor Overview</summary>

This PR adds a new About me Button, (Will replace the old one, for now uses it for reference.) and the background **"About Me" system**, mainly the **ssrpmanagement.dm subsystem (ServerSide)**, and **aboutme_core.dm, and aboutme_record.dm (Player-Facing)** as center points. All supporting files are in the aboutme folder in the modulartfn folder.

There are two places that this directly lives in other places in the code: (Both are VITAL)
- mob/human/ComponentInitialize() ---> AddComponent(/datum/component/about_me)

and for the TGUI,
- "The-Final-Nights\tgui\packages\tgui\interfaces\AboutmeInt.jsx"

- `aboutme_core.dm`: Component logic, player character access to UI, and for immersive access the system.
- `aboutme_record.dm`: Bare bones "character's account." Assembles the character's setup and preferences, and their role choice on round join for the round and puts it together into something usable by the system. Boiled down even further... it's basically a fancy datum id key holder that grabs display information from the subsystem's stored datums.
- `aboutme_tgui.dm`: Handles the TGUI about me interface for each component. Supporting files are the branching options.
- `aboutme_defines.dm`: Group, Chronicle, Relationship, Memory types, and other supporting defines.
- `group.dm`, `relationships.dm`, `memory.dm`, `chronicle.dm`: Core data objects.
- `ssrpmanagement.dm`: Canonical groups/ and future record initialization, central GLOB manager for About-Me.
- TGUI: `AboutmeInt.jsx` added, Main About me Interface Window, everything else is TGUI Input prompts, or single use data windows.
- Component attached to `/mob/living/carbon/human/ComponentInitialize()`

</details>

---

## 🗒️ Changelog

:cl: MichaelEUkari  
add: New "About Me" panel for characters, with Overview, Groups, Relationships, Chronicle, and Memory tabs
add: Group management with roles, leaders, officers, and custom factions
add: Memory and relationship systems that evolve with roleplay
add: Chronicle system for shared group events
add: SSRPmanagement subsystem for RP record keeping
admin: Group edits, kicks, and logs available via backend tools
server: Character data is saved per slot; unclaimed data fades over time 
/:cl:

nikothedude commented:
peak has arrived

XeonMations commented:
Please ensure this PR follows the [modularization guide](https://github.com/The-Final-Nights/The-Final-Nights/blob/master/modular_tfn/readme.md), [coding standards ](https://github.com/The-Final-Nights/The-Final-Nights/blob/master/.github/guides/STANDARDS.md) and the [style guide](https://github.com/The-Final-Nights/The-Final-Nights/blob/master/.github/guides/STYLE.md)

ChristopherDAdams commented:
> Please ensure this PR follows the [modularization guide](https://github.com/The-Final-Nights/The-Final-Nights/blob/master/modular_tfn/readme.md), [coding standards ](https://github.com/The-Final-Nights/The-Final-Nights/blob/master/.github/guides/STANDARDS.md) and the [style guide](https://github.com/The-Final-Nights/The-Final-Nights/blob/master/.github/guides/STYLE.md)

Alright for the Modular guide, I've fixed up my ReadMe, as well as added my "// THIS IS A NOVA SECTOR UI FILE" to AboutmeInt.jsx. Added // NOVA EDIT ADDITION, to the external component initialization in human. So, externally, all settled.
Everything is in a single folder in modular tfn otherwise, I put about_me.dm in _DEFINES/tfn folder. 

In regards to the Code Standards, the PR is very much object oriented, with safe fall backs. Using as many good practices as i could think of with my experience.

And the style is consistent throughout. It could use a few more defines, but I have so many already and the code handles keys hosted in GLOB and the subsystem, so that handles dynamic generation. The places that need defines, are meant to be variable.

Right now testing groups and getting feedback on that is the major thing. Then Chronicles, Relationships, and Memories, will be bare bones for now to be used as tools for players and groups.

I'll be finishing up some more tgui buttons, and finishing the species mounting, then finally creating organizations for the entire city. Then it should be ready for testing on the test server, and potential test merges.

ChristopherDAdams commented:
I believe I'm ready for my baby to get ripped to shred. <3

XeonMations commented on modular_tfn/modules/roleplay_management/0_ssroleplay_management_core.dm:
```diff
@@ -0,0 +1,154 @@
+	..()
+	for (var/group_id in GLOB.groups)
+		var/datum/group/G = GLOB.groups[group_id]
+		if (!G?.active_votes || !length(G.active_votes)) continue
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+/// Sets the character's goals string via the aboutme_record.
+/datum/component/about_me/proc/set_goals(goals)
+	var/datum/aboutme_record/R = get_aboutme_record()
+	if (R) R.set_goals(goals)
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+	if (!istype(mob, /mob/living/carbon/human)) return
+	var/mob/living/carbon/human/H = mob
+	var/datum/component/about_me/C = H.GetComponent(/datum/component/about_me)
+	if (!C) return
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+/client/verb/DebugAboutMePayload()
+	set name = "About Me Debug Payload"
+	set category = "IC"
+	if (!istype(mob, /mob/living/carbon/human)) return
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+	var/mob/living/carbon/human/H = mob
+	var/datum/component/about_me/C = H.GetComponent(/datum/component/about_me)
+	if (!C) return
+	to_chat(src, "<span class='notice'>[json_encode(C.get_full_payload(), TRUE)]</span>")
```
Span macros.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+/// - Used for in-group leadership votes or officer promotions.
+/// - Registers the player's choice and updates the vote tally.
+/datum/component/about_me/proc/prompt_vote_on_group(datum/group_vote/V)
+	if (!V || !owner || !ismob(owner)) return
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+/datum/component/about_me/proc/prompt_vote_on_group(datum/group_vote/V)
+	if (!V || !owner || !ismob(owner)) return
+	var/datum/group/G = SSroleplay_management.get_group_by_key(V.group_id)
+	if (!G || V.has_voted(character_key)) return
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+	var/title = "Vote in [G.name]: [V.vote_type]"
+	var/message = "Do you vote YES or NO to [V.vote_type] [target]?"
+	var/choice = tgui_input_list(owner, message, title, list("Yes", "No"))
+	if (isnull(choice)) return
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+	if (!V || !owner || !ismob(owner)) return
+	var/datum/group/G = SSroleplay_management.get_group_by_key(V.group_id)
+	if (!G || V.has_voted(character_key)) return
+	var/target = G.member_names[V.target_character_key] || V.target_character_key
+	var/title = "Vote in [G.name]: [V.vote_type]"
+	var/message = "Do you vote YES or NO to [V.vote_type] [target]?"
+	var/choice = tgui_input_list(owner, message, title, list("Yes", "No"))
```
```suggestion
	var/target = G.member_names[V.target_character_key] || V.target_character_key
	var/choice = tgui_input_list(owner, "Do you vote YES or NO to [V.vote_type] [target]?", "Vote in [G.name]: [V.vote_type]", list("Yes", "No"))
```

ChristopherDAdams replied:
Took this changes and added them.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_core.dm:
```diff
@@ -0,0 +1,172 @@
+	var/choice = tgui_input_list(owner, message, title, list("Yes", "No"))
+	if (isnull(choice)) return
+	V.add_vote(character_key, choice == "Yes")
+	to_chat(owner, "<span class='notice'>Your vote for [target] has been recorded.</span>")
```
Span macros.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+			general["armory_code"] = armory.pincode
+		var/obj/keypad/panic_room/panic = find_keypad(/obj/keypad/panic_room)
+		if (panic && (role in list("Prince")))
+			general["panic_room_code"] = panic.pincode
```
seneschal also gets the pin to the panic room

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+		"goals"           = edit_goals || "",
+		"personal_quote"  = edit_personal_quote || ""
+	)
+	// --- Role-specific codes Everyone gets a bank account code, then roles for bank, panic room, armory, etc
```
surely theres an easier way of doing this than doing a for loop for everyone's bank accounts?

ChristopherDAdams replied:
Sadly that's the only way it works right now. There's a GLOBAL_LIST_EMPTY(bank_account_list) empty, but its not set up right now and im trying to stay in scope.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+
+/// Builds a dictionary of overview data for the panel's Overview tab.
+/// Pulls from both saved fields and live mob state (name, species, stats, codes, etc).
+/datum/aboutme_record/proc/build_overview_data(mob/living/carbon/human/owner)
```
Why is this proc needed even? Why not just shove it in static_ui_data for the TGUI menu?

ChristopherDAdams replied:
Changed it to get_ui_overview_data() to be sure to keep it within intent. Still needs work. Right now it literately ties all the character's overview information into two lists, general information which applies to everyone, and species, which only applies and is displayed to non-humans. I split the species procs into different switch cases and made them independent procs for more clarity. Working through style throughout still as well.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+/// Assembles and returns the full About Me payload for the TGUI panel.
+/// Contains overview, groups, relationships, chronicle, and categorized memories.
+/// @param owner The mob/living associated with this record.
+/datum/aboutme_record/proc/update_payload(mob/living/carbon/human/owner)
```
I highly doubt we need to be giving this much information to the ui every time its opened.

ChristopherDAdams replied:
/datum/aboutme_record/proc/update_payload(mob/living/carbon/human/owner)
	return list(
		"overview" = get_ui_overview_data(owner),
		"groups" = get_ui_groups(owner),
		"relationships" = get_ui_relationships(owner),
		"chronicle" = get_ui_chronicles(owner),
		"memories" = get_ui_memories_by_tag(owner)
	)
Changed my naming structure for now to bring it all into the same format and make the intention of the procs clearer, cut them down a bunch as well. Returns 5 Lists, for 5 datum types, for display in 5 Tabs. Shared in the subsystem, accessed by id and character information.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+	general["stats"] = stats
+
+	// --- Species-specific details (Kindred, Garou, Ghoul, etc)
+	var/list/species_block = list(
```
...what? Why is this here?

ChristopherDAdams replied:
Fixed it with the switch cases, as below.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+
+
+	// Kindred
+	if (iskindred(owner))
```
use a switch case of each of these.

ChristopherDAdams replied:
switch (species_type)
		if (/datum/species/kindred)
			return list("general" = general, "species" = get_species_ui_kindred(owner))
		if (/datum/species/garou)
			return list("general" = general, "species" = get_species_ui_garou(owner))
		if (/datum/species/ghoul)
			return list("general" = general, "species" = get_species_ui_ghoul(owner))
		if (/datum/species/kuei_jin)
			return list("general" = general, "species" = get_species_ui_cathayan(owner))
	//more species cases as needed
	// Fallback/default
	return list("general" = general, "species" = get_species_ui_default(owner))
	
	done.

ChristopherDAdams replied:
with the bonus if these are fundamentally changed it will easily lead back.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+/datum/aboutme_record/proc/group_by_type(list/groups)
+	var/list/by_type = list()
+	for (var/G in groups)
+		if (!islist(G)) continue
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+	for (var/G in groups)
+		if (!islist(G)) continue
+		var/type = G["type"]
+		if (!istext(type) || !length(type)) type = "unknown"
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+		if (!islist(G)) continue
+		var/type = G["type"]
+		if (!istext(type) || !length(type)) type = "unknown"
+		if (!(type in by_type)) by_type[type] = list()
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+/// Returns all groups this character is a member of, organized by type, and formatted for UI.
+/datum/aboutme_record/proc/get_ui_groups(mob/living/carbon/human/owner)
+	var/list/group_objects = list()
+	for (var/group_key in src.group_keys)
```
You dont need ``src.`` here

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+	var/list/group_objects = list()
+	for (var/group_key in src.group_keys)
+		var/datum/group/G = SSroleplay_management.get_group_by_key(group_key)
+		if (!G) continue
```
Style guide.

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+/// Returns all relationships for this character that are visible to the viewer, formatted for UI.
+/datum/aboutme_record/proc/get_ui_relationships(mob/living/carbon/human/owner)
+	var/list/output = list()
+	for (var/key in src.relationship_keys)
```
you dont need ``src.`` here

XeonMations commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+	var/list/output = list()
+	for (var/key in src.relationship_keys)
+		var/datum/relationships/R = SSroleplay_management.get_relationship_by_key(key)
+		if (!R || !R.visible) continue
```
Style guide.

buffyuwu commented on modular_tfn/modules/roleplay_management/about_me_tgui.dm:
```diff
@@ -0,0 +1,99 @@
+	try
+		return get_full_payload()
+	catch (var/exception/e)
+		return list("error" = "ui_data exception: [e]")
```
holy vibecode

XeonMations commented on tgui/packages/tgui/interfaces/AboutmeInt.jsx:
``.tsx``, not ``.jsx``

ChristopherDAdams commented on modular_tfn/modules/roleplay_management/about_me_tgui.dm:
```diff
@@ -0,0 +1,99 @@
+//BUTTONS! This is where the UI goes to aboutme_tgui_player_input.dm
+/datum/component/about_me/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if (.) return
```
If its not needed, right now, I'm taking it out! All over.

ChristopherDAdams commented on modular_tfn/modules/roleplay_management/about_me_record.dm:
```diff
@@ -0,0 +1,311 @@
+/// Sets the character's gender field.
+/datum/aboutme_record/proc/set_gender(n)              edit_gender = n
+/// Sets the character's physical description field.
+/datum/aboutme_record/proc/set_physical_desc(n)       edit_physical_desc = n
```
Kind of a pointless setter anyway but going through and fixing absolutely all of my magic vars to be more descriptive and break that habit.

ChristopherDAdams commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -36,6 +36,10 @@
 	. = ..()
 	if(!CONFIG_GET(flag/disable_human_mood))
 		AddComponent(/datum/component/mood)
+	if(!GetComponent(/datum/component/about_me))
+		// TFN EDIT ADDITION START - ABOUT_ME_CORE
+		AddComponent(/datum/component/about_me)
+		// TFN EDIT ADDITION END
```
Done.

hex37 commented on modular_tfn/modules/roleplay_management/code/ssroleplay_management_groups.dm:
```diff
@@ -0,0 +1,458 @@
+//   - Runtime group lookup and modification
+// ================================================================
+//Species Factions, and Base clan membership is already handled before here.
+/datum/controller/subsystem/roleplay_management/var/list/ROLE_GROUP_MAPPINGS = list(
```
You should load this list dynamically at the subsystem's Initialize with typesof() instead of doing this hardcoded method.

hex37 commented on modular_tfn/modules/roleplay_management/code/ssroleplay_management_groups.dm:
```diff
@@ -0,0 +1,458 @@
+
+
+// ---------------- INITIALIZATION ----------------
+/datum/controller/subsystem/roleplay_management/Initialize()
```
Only call functions in one place if at all possible, especially within a module; if you do function overwriting throughout your code it becomes near-impossible to follow.

FalloutFalcon commented on modular_tfn/modules/roleplay_management/code/about_me/about_me_tgui_player_chronicle.dm:
```diff
@@ -0,0 +1,156 @@
+	if (isnull(choice) || choice == "Back") return
+
+	switch(choice)
+		if ("Manage Personal Chronicle") return src.prompt_manage_personal_chronicle(user)
+		if ("Manage Group Chronicle")    return src.prompt_manage_group_chronicle(user)
+
```
The standard in TG code is the not put these on the same line (your doing this ALOT in this pr right now). You also used space instead of tab which is evil.
```suggestion
		if ("Manage Personal Chronicle")
			return src.prompt_manage_personal_chronicle(user)
		if ("Manage Group Chronicle")
			return src.prompt_manage_group_chronicle(user)

```

FalloutFalcon commented on tgui/packages/tgui/interfaces/AboutmeInt.tsx:
its a draft but would recommend breaking this tsx file up alot (I would go look at tg for how they sort them), your also missing types for everything despite being type script.
Also css is generally handled inside of scss files for example `tgui\packages\tgui\styles\interfaces\Roulette.scss`

ChristopherDAdams commented on modular_tfn/master_files/code/modules/mob/living/carbon/human/human_defines.dm:
```diff
@@ -5,3 +5,8 @@
 	var/ooc_notes
 	var/show_flavor_text_when_masked
+
+/mob/living/carbon/human/ComponentInitialize()
+	. = ..()
+	if(!GetComponent(/datum/component/about_me))
+		AddComponent(/datum/component/about_me)
```
True, I asked Xeon if i could tuck it in there, i could make another file for it, but seems a bit like that file should be called human_overrides, to keep the modular tfn human overrides together maybe? I'll check with Xeon for what to do with that.
